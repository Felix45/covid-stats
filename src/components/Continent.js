import React from 'react';
import PropTypes from 'prop-types';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import { NavLink } from 'react-router-dom';

const Continent = (props) => {
  const { region } = props;
  return (
    <Col xs={6} className={`m-0 p-0 continent ${region.name.toLowerCase()}`}>
      <Card>
        <NavLink to={`/continent/${region.name}`}>
          <Card.Img className="pt-2" src={`${process.env.PUBLIC_URL}/images/${region.image}`} />
        </NavLink>
        <Card.Body className="text-end mb-2">
          <div className="mx-3">
            <h2 className="fw-bold">{region.name}</h2>
          </div>
          <div className="fw-bold mx-3">
            <strong>
              Cases
              <br />
              { region.cases.toLocaleString('en-US') }
            </strong>
          </div>
        </Card.Body>
      </Card>
    </Col>
  );
};

Continent.propTypes = {
  region: PropTypes.instanceOf(Object).isRequired,
};

export default Continent;
