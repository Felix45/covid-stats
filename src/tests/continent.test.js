import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux';
import Continent from '../components/Continent';
import continents from '../mock/data';

describe('Test single continent renders correctly', () => {
  test('Test single continent view renders correctly', () => {
    const Tree = renderer.create(
      <Provider store={store()}>
        <BrowserRouter>
          <Continent region={continents[0]} />
        </BrowserRouter>
      </Provider>,
    );
    expect(Tree).toMatchSnapshot();
  });
});
