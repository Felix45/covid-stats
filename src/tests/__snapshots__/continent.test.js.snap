// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test single continent renders correctly Test single continent view renders correctly 1`] = `
<div
  className="m-0 p-0 continent africa col-6"
>
  <div
    className="card"
  >
    <a
      className=""
      href="/continent/Africa"
      onClick={[Function]}
    >
      <div
        className="p-2 text-end card-header"
      >
        <span
          className="fa fa-arrow-circle-right fa-lg"
        />
      </div>
      <img
        className="card-img pt-2"
        src="/images/africa.png"
      />
    </a>
    <div
      className="text-end mb-2 card-body"
    >
      <div
        className="mx-3"
      >
        <h2
          className="fw-bold"
        >
          Africa
        </h2>
      </div>
      <div
        className="fw-bold mx-3"
      >
        <strong>
          Cases
          <br />
          0
        </strong>
      </div>
    </div>
  </div>
</div>
`;
